developer-hub/src/polkadot_sdk/substrate.rs

```diff
@@ -0,0 +1,150 @@
+//!   framework is using for building the runtime. Notable examples are [`sp_api`] and [`sp_io`],
+//!   which form the communication bridge between the client and runtime.
+//! * `pallet-*` and `frame-*` crates, located under `./frame` folder. These are the crates related
+//!   to FRAME. See [`frame`] for more information.
+//!
+//! ### Wasm Build
+//!
+//! Many of the Substrate crates, such as entire `sp-*`, need to compile to both Wasm (when a Wasm
+//! runtime is being generated) and native (for example, when testing). To achieve this, Substrate
+//! follows the convention of the Rust community, and uses a `feature = "std"` to signify that a
+//!  crate is being built with the standard library, and is built for native. Otherwise, it is built
+//!  for `no_std`.
+//!
+//! This can be summarized in `#![cfg_attr(not(feature = "std"), no_std)]`, which you can often find
+//! in any Substrate-based runtime.
+//!
+//! Substrate-based runtimes use [`substrate_wasm_builder`] in their `build.rs` to automatically
+//! build their Wasm files as a part of normal build commandsOnce built, the wasm file is placed in
+//! `./target/{debug|release}/wbuild/{runtime_name}.wasm`.
+//!
+//! ### Binaries
+//!
+//! Multiple binaries are shipped with substrate, the most important of which are located in the
+//! [`./bin`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/bin) folder.
+//!
+//! * [`node_cli`] is an extensive substrate node that contains the superset of all runtime and
+//!   client side features. The corresponding runtime, called [`kitchensink_runtime`] contains all
+//!   of the modules that are provided with `FRAME`. This node and runtime is only used for testing
+//!   and demonstration.
+//!     * [`chain_spec_builder`]: Utility to build more detailed chain-specs for the aforementioned
+//!       node. Other projects typically contain a `build-spec` subcommand that does the same.
+//! * [`node_template`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/bin/node-template):
+//!   a template node that contains a minimal set of features and can act as a starting point of a
+//!   project.
+//! * [`subkey`]: Substrate's key management utility.
+//!
+//! ### Anatomy of a Binary Crate
+//!
+//! From the above, [`node_cli`]/[`kitchensink_runtime`] and `node-template` are essentially
+//! blueprints of a substrate-based project, as the name of the latter is implying. Each
+//! substrate-based project typically contains the following:
+//!
+//! * Under `./runtime`, a `./runtime/src/lib.rs` which is the top level runtime amalgamator file.
+//!   This file typically contains the [`frame::runtime::prelude::construct_runtime`] and
+//!   [`frame::runtime::prelude::impl_runtime_apis`] macro calls, which is the final definition of a
+//!   runtime.
+//!
+//! * Under `./node`, a `main.rs`, which is the point, and a `./service.rs`, which contains all the
```

Comment:
Maybe 'starting point'?
```suggestion
//! * Under `./node`, a `main.rs`, which is the starting point, and a `./service.rs`, which contains all the
```